<p>Inside complex code parts (for exemple multiple loops, complex data constructions...), avoid using finally in try/catch. It will save CPU cycles consumption.</p>
<h2>Noncompliant Code Example</h2>
<pre>
try
{
  $picture = PDF_open_image_file($PDF, "jpeg", $imgFile, "", 0); // This is the original statement, this works on PHP4
}
catch(Exception $ex)
{
  $msg = "Error opening $imgFile for Product $row['Identifier']";
  throw new Exception($msg);
}
finally {
  $msg = "bloc finally";
}

</pre>
<h2>Compliant Solution</h2>
<pre>
//try
if (file_exists($imgFile)) {
    $picture = PDF_open_image_file($PDF, "jpeg", $imgFile, "", 0);
}

//catch
if (!$picture) {
   $msg = "Error opening $imgFile for Product $row['Identifier']";
   print $msg;
}
//finally
print $msg;
</pre>
